.github/workflows/test.yml

```diff
@@ -0,0 +1,46 @@
+name: Test
+on:
+  workflow_dispatch:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  script-method:
+    name: Test plugin
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download yt-dlp
+        run: curl -L https://github.com/yt-dlp/yt-dlp-nightly-builds/releases/latest/download/yt-dlp -o yt-dlp && chmod +x ./yt-dlp
+      - name: Install and build server dependencies
+        run: cd server/ && yarn install --frozen-lockfile && npx tsc
+      - name: Install plugin
+        run: |
+          mkdir ~/yt-dlp-plugins && curl -L "https://github.com/coletdjnz/yt-dlp-get-pot/releases/download/v0.1.1/yt-dlp-get-pot.zip" -o ~/yt-dlp-plugins/yt-dlp-get-pot.zip
+          cp -aT plugin/ ~/yt-dlp-plugins/bgutil-ytdlp-pot-provider/
+      - name: Test script method
+        run: |
+          script_response=$(./yt-dlp -vF --print-traffic BaW_jenozKc --extractor-args "youtube:getpot_bgutil_script=server/build/generate_once.js" 2>&1)
+          echo $script_response
+          if [[ "$script_response" != *"BgUtilScriptPot: _get_pot_via_script response"* ]]; then
+            echo "BgUtilScriptPot was not invoked"
+            exit 1
+          fi
+      - name: Test server method
+        run: |
+          cd server/
+          node build/main.js&
+          cd ..
+          until curl --silent --fail http://127.0.0.1:4416/ping; do
+            echo "Waiting for server to be up..."
+            sleep 5
+          done
+          script_response=$(./yt-dlp -vF --print-traffic BaW_jenozKc --extractor-args "youtube:getpot_bgutil_baseurl=http://127.0.0.1:4416" 2>&1)
+          echo $script_response
+          if [[ "$script_response" != *"BgUtilHTTPPot: Generating POT via HTTP server"* ]]; then
+            echo "BgUtilHTTPPot was not invoked"
```

Comment:
consider using github format

Reply:
see: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-error-message